dnsseed.go

```diff
@@ -78,10 +78,15 @@ func creep() {
 					p.NA().IP.String(), msg.Services, msg.SubnetworkID)
 				// Mark this peer as a good node.
 				amgr.Good(p.NA().IP, msg.Services, msg.SubnetworkID)
-				// Ask peer for some addresses.
-				p.QueueMessage(wire.NewMsgGetAddr(true, nil), nil)
-				// notify that version is received and Peer's subnetwork ID is updated
-				onVersion <- struct{}{}
+
+				// The following is in a separate goroutine to avoid a deadlock in which
+				// OnVersion blocks AssociateConnection and AssociateConnection blocks QueueMessage.
+				spawn(func() {
+					// Ask peer for some addresses.
+					p.QueueMessage(wire.NewMsgGetAddr(true, nil), nil)
+					// notify that version is received and Peer's subnetwork ID is updated
+					onVersion <- struct{}{}
```

Comment:
Do we really need onVersion channel once AssociateConnection is sync?

It seems like it's only used to check for timeout of the handshake.
Maybe instead, have a goroutine in `creep` that calls AssociateConnection and once done pings a local channel?